ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -166,23 +167,35 @@ impl InternalIndexerDBService {
 
     pub async fn run(&mut self, node_config: &NodeConfig) -> Result<()> {
         let mut start_version = self.get_start_version(node_config).await?;
+        let mut is_caught_up = false;
 
         loop {
-            let start_time: std::time::Instant = std::time::Instant::now();
-            let next_version = self.db_indexer.process_a_batch(start_version)?;
-
-            if next_version == start_version {
+            let mut start_time: std::time::Instant = std::time::Instant::now();
+            if is_caught_up {
                 if let Ok(recv_res) =
                     tokio::time::timeout(Duration::from_millis(100), self.update_receiver.changed())
                         .await
                 {
-                    if recv_res.is_err() {
-                        info!("update sender is dropped");
-                        return Ok(());
+                    match recv_res {
+                        Ok(()) => {
+                            start_time = *self.update_receiver.borrow();
+                        },
+                        Err(_) => {
+                            info!("update sender is dropped");
+                            return Ok(());
+                        },
                     }
                 }
+            }
+            let next_version = self.db_indexer.process_a_batch(start_version)?;
+
+            if next_version == start_version {
+                is_caught_up = true;
                 continue;
```

Comment:
shouldn't we log the latency always?

Reply:
no, next_version == start_version means no data was written to internal indexer so there is nothing done in this step.

Reply:
okay.. it's the reverse -- we should log only when it's caught up to be accurate